From c69e2157c41d364efea5e4b6350dd6ead86c3c4f Mon Sep 17 00:00:00 2001
From: Jakub Jirutka <jakub@jirutka.cz>
Date: Sat, 20 May 2017 21:44:42 +0200
Subject: [PATCH 08/10] Don't fetch grpc gem from RubyGems

Grpc gem is total shit, doesn't build from source without patching,
embeds three third-party libs including Google's very own fork of
OpenSSL and root CAs, contains a lot of bloat etc.

Therefore I install this shit from my Alpine package and
copy it into vendor/gems/ before running bundler.
---
 Gemfile      |  3 +++
 Gemfile.lock | 10 +++++++---
 2 files changed, 10 insertions(+), 3 deletions(-)

diff --git a/Gemfile b/Gemfile
index 64385c0..32da951 100644
--- a/Gemfile
+++ b/Gemfile
@@ -392,6 +392,9 @@ gem 'sys-filesystem', '~> 1.1.6', group: :exclude
 # Gitaly GRPC client
 gem 'gitaly', '~> 0.14.0'
 
+# This Google gem is totally messed, we must handle it specially.
+gem 'grpc', path: 'vendor/gems/grpc', require: false
+
 gem 'toml-rb', '~> 0.3.15', require: false
 
 # Feature toggles
diff --git a/Gemfile.lock b/Gemfile.lock
index c607c7e..54649f9 100644
--- a/Gemfile.lock
+++ b/Gemfile.lock
@@ -1,3 +1,9 @@
+PATH
+  remote: vendor/gems/grpc
+  specs:
+    grpc (1.3.2)
+      google-protobuf (~> 3.1)
+
 GEM
   remote: https://rubygems.org/
   specs:
@@ -352,9 +358,6 @@ GEM
     grape-entity (0.6.0)
       activesupport
       multi_json (>= 1.3.2)
-    grpc (1.4.0)
-      google-protobuf (~> 3.1)
-      googleauth (~> 0.5.1)
     haml (4.0.7)
       tilt
     haml_lint (0.21.0)
@@ -988,6 +991,7 @@ DEPENDENCIES
   google-api-client (~> 0.8.6)
   grape (~> 0.19.0)
   grape-entity (~> 0.6.0)
+  grpc!
   haml_lint (~> 0.21.0)
   hamlit (~> 2.6.1)
   hashie-forbidden_attributes
-- 
2.10.1 (Apple Git-78)

